src/mpid/ch4/netmod/ofi/ofi_coll.h

```diff
@@ -12,20 +12,52 @@
 #define OFI_COLL_H_INCLUDED
 
 #include "ofi_impl.h"
+#include "ch4_coll_select.h"
+#include "ch4_coll_params.h"
+#include "ofi_coll_select.h"
 
 #undef FUNCNAME
 #define FUNCNAME MPIDI_NM_mpi_barrier
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-static inline int MPIDI_NM_mpi_barrier(MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag)
+static inline int MPIDI_NM_mpi_barrier(MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag,
+                                       void * ch4_algo_parameters_container_in)
```

Comment:
The algo params argument is a `void *` here but `MPIDI_OFI_coll_algo_container_t` in the POSIX function headers. Please make them consistent, preferably with the correct type to avoid casting immediately.